packages/graphql-language-service-server/src/MessageProcessor.js

```diff
@@ -9,409 +9,205 @@
+  );
 
-      query = cachedDocument.content.text;
-      result = await languageService.getDefinition(
-        query,
-        pos,
-        textDocument.uri,
-      );
-      const formatted = result
-        ? result.definitions.map(res => ({
-            uri: path.join('file://', res.path),
-            range: res.range,
-          }))
-        : [];
-      sendMessageIPC(
-        convertToRpcMessage({
-          id: message.id,
-          result: formatted,
-        }),
-        writer,
-      );
-      break;
-    case '$/cancelRequest':
-      if (!message.params || !message.params.id) {
-        // `id` is required.
-        return;
-      }
-      const requestIDToCancel = message.params.id;
-      const index = REQUEST_IDS_IN_PROGRESS.indexOf(requestIDToCancel);
-      if (index !== -1) {
-        REQUEST_IDS_IN_PROGRESS.splice(index, 1);
-        // A cancelled request still needs to send an empty response back
-        sendMessageIPC({id: requestIDToCancel}, writer);
-      }
-      break;
-    case 'shutdown':
-      // prepare to shut down the server
-      break;
+  const cachedDocument = textDocumentCache.get(documentUri);
+  if (!cachedDocument) {
+    return;
   }
+
+  // Send the diagnostics onChange as well
+  const diagnostics = await provideDiagnosticsMessage(
+    cachedDocument.content.text,
+    documentUri,
+  );
+
+  return {documentUri, diagnostics};
 }
 
-export async function processStreamMessage(
-  message: string,
-  configDir: ?string,
+export async function handleDidCloseNotification(
+  params: NotificationMessage,
 ): Promise<void> {
-  if (message.length === 0) {
-    return;
+  // For every `textDocument/didClose` event, delete the cached entry.
+  // This is to keep a low memory usage && switch the source of truth to
+  // the file on disk.
+  if (!params || !params.textDocument) {
+    // `textDocument` is required.
+    throw new Error('`textDocument` is required.');
   }
-  if (!graphQLCache) {
-    const graphQLConfigDir = findGraphQLConfigDir(
-      configDir ? configDir.trim() : process.cwd(),
-    );
-    if (!graphQLConfigDir) {
-      process.stdout.write(
-        JSON.stringify(
-          convertToRpcMessage({
-            id: '-1',
-            error: {
-              code: ERROR_CODES.SERVER_NOT_INITIALIZED,
-              message: '.graphqlrc not found',
-            },
-          }),
-        ),
-      );
-      return;
-    }
-    graphQLCache = await getGraphQLCache(graphQLConfigDir);
+  const textDocument = params.textDocument;
+
+  if (textDocumentCache.has(textDocument.uri)) {
+    textDocumentCache.delete(textDocument.uri);
   }
-  if (!languageService) {
-    languageService = new GraphQLLanguageService(graphQLCache);
+}
+
+export async function handleInitializeRequest(
+  params: RequestMessage,
+  token: CancellationToken,
+  configDir?: string,
+): Promise<InitializeResult.type> {
+  if (!params || !params.rootPath) {
+    // `rootPath` is required
```

Comment:
Ah, just to throw an error when it's not there. It's an useful debugging "tool" for me